### PR TITLE
Improved plastic surgery; pick the target name

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
add: During the "reshape face" step of plastic surgery, the surgeon can
pick the new name that the person undergoing surgery will gain. Screwing
up will give them a random name instead.
del: Plastic surgery will no longer fix disfigurement.
/:cl:

What horrors will unscrupulous surgeons do to people? Will Abductors
just start renaming all of their subjects to "Subject Y", "Subject Z", "Subject Ran-out-of-letters"

### CODE REVIEW

**Comment by deathride58** [comment]
You can probably make it so it has a solid chance to change one single character in the name, to prevent any !!!fun!!! that may occur from surgeons naming all of their patients the same exact thing, while also adding the fun of "Cap, did you mean Dimitri Petrovski, Dimitrl Petrovski, Dimitri Petyovski, Or Dinitri Petrovoski?"

File: code/modules/surgery/plastic_surgery.dm [hunk 1]
```diff
@@ -8,20 +8,29 @@
...
-	if(target.has_trait(TRAIT_DISFIGURED, TRAIT_GENERIC))
-		target.remove_trait(TRAIT_DISFIGURED, TRAIT_GENERIC)
-		user.visible_message("[user] successfully restores [target]'s appearance!", "<span class='notice'>You successfully restore [target]'s appearance.</span>")
-	else
-		var/oldname = target.real_name
-		target.real_name = target.dna.species.random_name(target.gender,1)
-		var/newname = target.real_name	//something about how the code handles names required that I use this instead of target.real_name
-		user.visible_message("[user] alters [oldname]'s appearance completely, [target.p_they()] is now [newname]!", "<span class='notice'>You alter [oldname]'s appearance completely, [target.p_they()] is now [newname].</span>")
+	var/oldname = target.real_name
+	user.visible_message("[user] alters [oldname]'s appearance completely, [target.p_they()] is now [chosen_name]!", "<span class='notice'>You alter [oldname]'s appearance completely, [target.p_they()] is now [chosen_name].</span>")
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/surgery/plastic_surgery.dm]: are

Comment (Bawhoppen) [reply code/modules/surgery/plastic_surgery.dm]: What does p_they() actually do that just saying they doesn't?

Comment (deathride58) [reply code/modules/surgery/plastic_surgery.dm]: p_they() takes into account the mob's gender.

there has to be a p_are() to make the sentence grammatically correct no matter the mob's gender.

**Comment by Time-Green** [comment]
Maybe have them have a picture of someone in their hand

File: code/modules/surgery/plastic_surgery.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/surgery/plastic_surgery.dm]: just default to "they are now [name]"

Comment (ExcessiveUseOfCobblestone) [reply code/modules/surgery/plastic_surgery.dm]: wait is p_they = he/she/it?

**Comment by PKPenguin321** [comment]
>a male spaceman can change his name to a female one
>but can't actually get gender reassignment

Disgusting & transphobic. :angry:

**Comment by ExcessiveUseOfCobblestone** [comment]
having to rp a trap, heavens forbid not on my christian server!

**Comment by AlexTheSysop42** [comment]
@deathride58 characters with the same name can already be accomplished with genetics so it's not like it should cause too many issues

**Comment by KorPhaeron** [comment]
I think being able to set your name to whatever you want by applying a fireaxe to someones face is a little weird

Impersonating the captain should be a bit harder than that

**Comment by KorPhaeron** [comment]
I agree it would be cool for abductors to make creepy experiment names but I think it will mostly just be used with a mirror for pseudo ling or voice changer

**Comment by PKPenguin321** [comment]
I mean kor it requires surgery which requires two conspirators

**Comment by KorPhaeron** [comment]
Not hard for team antags, of which we have many

**Comment by coiax** [comment]
I mean, you can't die your hair with a mirror.

And clearly we should make changing your hairstyles require a BARBER. ;)

**Comment by MimicFaux** [comment]
How about the best of both worlds, give a prompt to choose from a list of generated names. Randomized, but you can at least pick the one you like the most.

**Comment by Bawhoppen** [comment]
@coiax We do need it

**Comment by Cyberboss** [comment]
kor don liek it

**Comment by coiax** [comment]
Kor should close it himself then. :P

**Comment by Bawhoppen** [comment]
Do you think the king has time to deal with peasants like us?

**Pull request closed without merging**